math/vc_aos/include/algebra/math/impl/vc_aos_vector.hpp

```diff
@@ -102,8 +136,11 @@ ALGEBRA_HOST_DEVICE constexpr auto eta(const vector_t &v) noexcept {
 /// @return a vector representing the cross product
 template <algebra::concepts::vc_aos_vector vector_t1,
           algebra::concepts::vc_aos_vector vector_t2>
-ALGEBRA_HOST_DEVICE constexpr auto cross(const vector_t1 &a, const vector_t2 &b)
-    -> decltype(a * b - b * a) {
+requires(std::same_as<algebra::traits::value_t<vector_t1>, double> ||
+         std::same_as<algebra::traits::value_t<vector_t2>, double>)
```

Comment:
Shouldn't they both be doubles?

Reply:
Indeed, I don't think we ever have a mixed precision case, but I thought it was safer to use the "negation" of the single precision case instead of potentially having no overload that fits

Reply:
Ah, right. But can float and double vectors be crossed with eachother?

Reply:
I guess we would see a bunch of double to single conversion warnings (as it probably should be atm). But the double implementation just rips every element of the simd vector out and calculates the cross product "the old fashioned way". So, I guess it should technically work

Reply:
Fair enough!